plugins/http_plugin/http_plugin.cpp

```diff
@@ -123,22 +163,132 @@ namespace eosio {
              };
             return handler;
          }
+         
+         bool is_unix_socket_address(const std::string& address) const {
+            return address[0] == '/' ||
+                   (address[0] == '.' &&
+                    (address[1] == '/' || (address.size() >= 3 && address[1] == '.' && address[2] == '/')));
+         }
 
-         void add_aliases_for_endpoint( const tcp::endpoint& ep, const string& host, const string& port ) {
-            auto resolved_port_str = std::to_string(ep.port());
-            plugin_state->valid_hosts.emplace(host + ":" + port);
-            plugin_state->valid_hosts.emplace(host + ":" + resolved_port_str);
+         bool on_loopback_only(const std::string& address) {
+            if (is_unix_socket_address(address))
+               return true;
+            auto [host, port] = split_host_port(address);
+            boost::system::error_code ec;
+            tcp::resolver             resolver(plugin_state->thread_pool.get_executor());
+            auto endpoints = resolver.resolve(host, port, boost::asio::ip::tcp::resolver::passive, ec);
+            EOS_ASSERT(!ec, chain::plugin_config_exception, "failed to resolve address: ${msg}", ("msg", ec.message()));
```

Comment:
Good point, should likely not throw from here.

Reply:
This looks like it is only called from `is_on_loopback` which is only currently called from `plugin_initialize`. However It seems like it would be better for this to only log a warning and return false. Seems like loopback should always be resolvable so unable to resolve likely means not on loopback.